workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx

```diff
@@ -3,117 +3,253 @@
-					</label>
-
-					<ClayButton
-						className="mb-1 ml-4"
-						displayType="secondary"
-						size="sm"
-					>
-						{i18n.translate('delete')}
-					</ClayButton>
-				</div>
+				<UploadLogo
+					onDeleteFile={() =>
+						dispatch({
+							payload: {
+								file: undefined,
+							},
+							type: SolutionTypes.SET_PROFILE,
+						})
+					}
+					onUpload={handleLogoUpload}
+					tooltip={ReactDOMServer.renderToString(
+						<span>
+							The icon is a small image representation of the app.
+							Icons must be a PNG, JPG, or GIF format and cannot
+							exceed 5MB. Animated images are prohibited. The use
+							of the Liferay logo, including any permitted
+							alternate versions of the Liferay logo, is permitted
+							only with Liferay&apos;s express permission. Please
+							refer to our{' '}
+							<a
+								href="https://www.liferay.com/trademark"
+								target="_blank"
+							>
+								trademark policy
+							</a>{' '}
+							for details.
+						</span>
+					)}
+					uploadedFile={file}
+				/>
 			</div>
 
-			<Form.Label className="mt-5" htmlFor="name" required>
+			<Form.Label
+				className="mt-5"
+				htmlFor="name"
+				info={tooltipInfo.name}
+				required
+			>
 				{i18n.translate('name')}
 			</Form.Label>
 
 			<Form.Input
 				name="name"
+				onChange={onChange}
 				placeholder="Enter solution name"
 				type="text"
 			/>
 
-			<Form.Label className="mt-5" htmlFor="description" required>
+			<Form.Label
+				className="mt-5"
+				htmlFor="description"
+				info={tooltipInfo.description}
+				required
+			>
 				{i18n.translate('description')}
 			</Form.Label>
 
 			<Form.Input
 				component="textarea"
 				name="description"
+				onChange={onChange}
 				placeholder="Enter solution description"
 				type="textarea"
 			/>
 
-			<div className="form-multiselect">
-				<Form.Label className="mt-5" htmlFor="categories" required>
-					{i18n.translate('categories')}
-				</Form.Label>
-
-				<ClayMultiSelect
-					inputName="description-selector"
-					items={items}
-					onChange={setValue}
-					onItemsChange={setItems}
-					sourceItems={sourceItems}
-					spritemap={getIconSpriteMap()}
-					value={value}
-				/>
+			{!isLoading && (
+				<div className="form-multiselect">
+					<Form.Label
+						className="mt-5"
+						htmlFor="categories"
+						info={tooltipInfo.categories}
+						required
+					>
+						{i18n.translate('categories')}
+					</Form.Label>
 
-				<Form.Label className="mt-5" htmlFor="tags" required>
-					{i18n.translate('tags')}
-				</Form.Label>
-
-				<ClayMultiSelect
-					inputName="tags-selector"
-					items={items}
-					onChange={setValue}
-					onItemsChange={setItems}
-					sourceItems={sourceItems}
-					spritemap={getIconSpriteMap()}
-					value={value}
-				/>
-			</div>
+					<ClayMultiSelect
+						{...{placeholder: 'Select categories'}}
+						inputName="description-selector"
+						items={categories}
+						key={'cat-' + categories.length}
```

Comment:
Could you use template string on this
```suggestion
						items={categories}
						key={`cat-${categories.length}`}
```